src/app/page.tsx

```diff
@@ -1,8 +1,14 @@
 "use client";
 import "leaflet/dist/leaflet.css";
-import HomepageMap from "@/components/Home/HomepageMap";
 import { StationsProvider } from "@/providers/StationsProvider";
+import dynamic from "next/dynamic";
 
+const HomepageMap = dynamic(
```

Comment:
out of curiosity, why changing this and importing it dynamically here?

Reply:
When I tried to build the app locally, the process failed due to a window object error I was facing before. HomepageMap component imports L from leaflet but NextJS doesn't like it, because it uses the window object. Originally I tried to dynamically import L but that failed. The next best thing was to dynamically import the parent component inside page.tsx.

Reply:
I'm still trying to understand the difference between the build-in mechanisms of next and react, but an alternative is to use the lazy() method of react like this:

```
import { lazy } from "react";

const HomepageMap = lazy(() => import("../components/Home/HomepageMap"));

```

Ref: https://react.dev/reference/react/lazy?fbclid=IwAR395CV-gm2qhbphPgSvdbwPaTtBi75H30Pcr_fh1uovmJThYadi-HShbDc

Reply:
Some differences I found so far:

> The difference between next/dynamic and React. lazy is that next/dynamic supports code splitting for server side rendered applications while React. lazy does not. next/dynamic is very similar to react-loadable with some modification for Next's purposes.

Small difference in import size:
![image](https://github.com/Virtuosofriend/fwm-app/assets/33863308/e1ee399c-1ed5-4f18-bd6d-ff7630e2dfa3)